sites/partners/cypress/e2e/default/03-listing.spec.ts

```diff
@@ -49,6 +49,26 @@ describe("Listing Management Tests", () => {
     cy.getByID("digitalApplicationChoice-error").contains("This field is required")
     cy.getByID("paperApplicationChoice-error").contains("This field is required")
     cy.getByID("referralOpportunityChoice-error").contains("This field is required")
+    // Verify the behavior of Exit discard & confirm
+    cy.contains("Listing Details").click()
+    cy.getByID("name").clear()
+    cy.getByID("name").type("Test - error messaging DISCARD")
+    cy.getByID("listingsExitButton").click()
+    cy.getByID("listing-save-before-exit-dialog-content").contains(
+      "Do you want to save your changes before you exit?"
+    )
+    cy.getByID("saveBeforeExitDiscard").click()
+    cy.contains("Test - error messaging")
+    cy.getByID("listingEditButton").contains("Edit").click()
+    cy.getByID("name").clear()
+    cy.getByID("name").type("Test - error messaging DISCARD")
+    cy.getByID("listingsExitButton").click()
```

Comment:
Could we use getByText for selectors like this where possible?

Reply:
Hmm, well for a simple string like "Exit" I'm not sure if we want to assume there's only one "Exit" on the page…?